debug/debug.go

```diff
@@ -14,8 +14,18 @@
 
 package debug
 
+import "context"
+
 // Logger is the interface used for debug logging. By default, it is unused.
+//
+// Deprecated: use ContextLogger instead.
```

Comment:
Let's say "Prefer ContextLogger" or similar. We'll keep letting people use Logger if they want.